packages/io/x3d-serializer/src/index.js

```diff
@@ -165,6 +168,24 @@ const convertGeom2 = (object, options) => {
   return group
 }
 
+/*
+ * generate attributes for Shape node
+ */
+
+const shapeAttributes = (object, attributes = {}) => {
+  if (object.name) {
+    Object.assign(attributes, { DEF: uniqueDefName(object.name) })
```

Comment:
The 'name' should be decided by the user. So, let's just use whatever the user provides.

Reply:
ok, I can remove the unique function. Would it be possible to provide a warning, perhaps to the console, in case the same name is used for multiple objects ?

Reply:
that sounds reasonable. by definition, are DEF (names) expected to be unique?